packages/web-components/src/components/ic-text-field/ic-text-field.tsx

```diff
@@ -140,122 +160,103 @@ export class TextField {
+   * If `true`, the value of the text field will have its spelling and grammar checked.
    */
-  @Prop() type: IcTextFieldTypes = "text";
-
+  @Prop() spellcheck: boolean = false;
   /**
-   * A hint to the browser for which keyboard to display.
-   * Possible values: `"none"`, `"text"`, `"tel"`, `"url"`,
-   * `"email"`, `"numeric"`, `"decimal"`, and `"search"`.
+   * @internal If `true`, an ellipsis will be displayed at the end of the value if the value is longer than the container.
    */
-  @Prop() inputmode: IcTextFieldInputModes = "text";
+  @Prop() truncateValue?: boolean;
 
   /**
-   * The name of the control, which is submitted with the form data.
+   * The type of control to display. The default type is text.
    */
-  @Prop() name: string = this.inputId;
+  @Prop() type: IcTextFieldTypes = "text";
 
   /**
-   * If `true`, the value of the text field will have its spelling and grammar checked.
+   * If `true`, the icon in input control will be displayed - only applies when validationStatus ='success'.
    */
-  @Prop() spellcheck: boolean = false;
+  @Prop() validationInline: boolean = false;
 
   /**
-   * @internal The active element when focus is on the ic-menu items.
-   */
-  @Prop() ariaActiveDescendant?: string;
-  /**
-   * @internal If `true`, an ellipsis will be displayed at the end of the value if the value is longer than the container.
+   *  @internal If `true`, the validation will display inline.
    */
-  @Prop() truncateValue?: boolean;
+  @Prop() validationInlineInternal: boolean = false;
 
   /**
-   * @internal Used to identify if the slotted menu is rendered
+   * The validation state - e.g. 'error' | 'warning' | 'success'.
    */
-  @Prop() ariaExpanded: string;
+  @Prop() validationStatus: IcInformationStatusOrEmpty = "";
 
   /**
-   * @internal Used to identify any related child component
+   * The validation state - e.g. 'error' | 'warning' | 'success'.
    */
-  @Prop() ariaOwns: string;
+  @Prop() validationText: string = "";
 
   /**
-   * @internal Used to identify whether inputting any text triggers more predictions
+   * The amount of time, in milliseconds, to wait to trigger the `icChange` event after each keystroke.
    */
-  @Prop() ariaAutocomplete: IcAriaAutocompleteTypes = undefined;
+  @Prop() debounce: number = 0;
 
-  /**
-   * @internal Used to set the role if not default textbox;
-   */
-  @Prop() role: string;
+  @Watch("debounce")
+  private debounceChanged() {
+    this.icChange = debounceEvent(this.icChange, this.debounce);
+  }
 
   /**
-   * @internal If `true`, the hidden form input will stop rendering for form submission.
+   * The value of the text field.
    */
-  @Prop() hiddenInput: boolean = true;
+  @Prop({ reflect: true, mutable: true }) value: string = "";
 
-  @State() numChars: number = 0;
-  @State() maxLengthExceeded: boolean = false;
-  @State() minValueUnattained: boolean = false;
-  @State() maxValueExceeded: boolean = false;
   @State() initialValue = this.value;
```

Comment:
Needs moving up to before the props

Reply:
this one is dependent on the value prop and will need eslint ignore when new rules incoporated